### PR TITLE
Fixing #37, #52, #53, #54, #55

### PR DESCRIPTION
**Pull request by erangatl**

### CODE REVIEW

File: docs/demo/general-form-elements.html [hunk 1]
```diff
@@ -312,15 +312,15 @@ <h4 class="sub-heading">Form Types</h4>
         <div data-lang="html">
     <form class="form-inline" role="form">
         <div class="form-group">
-            <label class="sr-only" for="exampleInputUsername1">Username</label>
-            <input type="text" class="form-control" id="exampleInputUsername1" placeholder="Enter Username">
```
Comment (thusithak) [inline docs/demo/general-form-elements.html]: can we refrain from using numerics in ids unless it's being auto generated?

File: docs/demo/general-form-elements.html [hunk 2]
```diff
@@ -138,14 +138,14 @@ <h4 class="sub-heading">Input groups</h4>
...
+                    <label class="sr-only" for="input-amount">Amount (in dollars)</label>
                     <div class="input-group">
                         <div class="input-group-addon">$</div>
-                        <input type="number" min="1" max="100000" class="form-control form-control-md" placeholder="Amount">
+                        <input id="input-amount" type="number" min="1" max="100000" class="form-control form-control-md" placeholder="Amount">
                         <div class="input-group-addon">.00</div>
                     </div>
                 </div>
-                <button type="submit" class="btn">Transfer cash</button>
+                <button id="btnTransfer" type="submit" class="btn">Transfer cash</button>
```
Comment (thusithak) [inline docs/demo/general-form-elements.html]: use same id naming convention

**Review by thusithak — CHANGES_REQUESTED** [review]

File: docs/demo/buttons.html [hunk 3]
```diff
@@ -4,71 +4,71 @@
...
-    <button type="button" class="btn btn-default">Default</button>
-    <button type="button" class="btn btn-primary">Primary</button>
-    <button type="button" class="btn btn-secondary">Secondary</button>
-    <button type="button" class="btn btn-success">Success</button>
-    <button type="button" class="btn btn-info">Info</button>
-    <button type="button" class="btn btn-warning">Warning</button>
-    <button type="button" class="btn btn-danger">Danger</button>
-    <button type="button" class="btn btn-link">Link</button>
-    <!-- /buttons -->
+            <!-- buttons -->
```
Comment (thusithak) [inline docs/demo/buttons.html]: can we affix or suffix some string to uniquely identify "btn-default" id from "btn-default" class, there shouldnt be an ambiguity.

**Review by thusithak — CHANGES_REQUESTED** [review]

File: docs/demo/general-form-elements.html (hunk 2 above)
Comment (erangatl) [inline docs/demo/general-form-elements.html]: Fixed

File: docs/demo/general-form-elements.html (hunk 1 above)
Comment (erangatl) [inline docs/demo/general-form-elements.html]: fixed

File: docs/demo/buttons.html (hunk 3 above)
Comment (erangatl) [inline docs/demo/buttons.html]: fixed

**Review by thusithak — APPROVED** [review]

File: docs/demo/buttons.html [hunk 4]
```diff
@@ -4,71 +4,71 @@
...
+            <button id="btn-default-id" type="button" class="btn btn-default">Default</button>
```
Comment (jeradrutnam) [inline docs/demo/buttons.html]: Why do we have id names ending with "*-id" ? Shall we take it out?

File: docs/demo/buttons.html [hunk 5]
```diff
@@ -4,71 +4,71 @@
...
     <h4 class="sub-heading">Button groups</h4>
-    
+
     <div class="message message-info">
         <h4><i class="icon fw fw-info"></i>Bootstrap</h4>
         <p>
-            For additional details please check bootstrap documentation: 
-            <a href="http://getbootstrap.com/components/#btn-groups" target="_blank">http://getbootstrap.com/components/#btn-groups</a>
+            For additional details please check bootstrap documentation:
+            <a id="btn-group-more" href="http://getbootstrap.com/components/#btn-groups" target="_blank">http://getbootstrap.com/components/#btn-groups</a>
```
Comment (jeradrutnam) [inline docs/demo/buttons.html]: Why do we have "btn-*" prefix for link id, shall we take it out?

File: docs/demo/general-form-elements.html [hunk 6]
```diff
@@ -16,46 +16,46 @@ <h4 class="sub-heading">Input Types</h4>
             <div class="col-md-6">
                 <div class="form-group">
                     <label class="control-label">Text</label>
-                    <input type="text" class="form-control" value="Some text">
+                    <input id="sample-text" type="text" class="form-control" value="Some text">
```
Comment (jeradrutnam) [inline docs/demo/general-form-elements.html]: Shall we add a prefix to these IDs? e.g. "form-control-*"

File: docs/demo/buttons.html (hunk 4 above)
Comment (erangatl) [inline docs/demo/buttons.html]: This is added to a review request to  uniquely identify "btn-default" id from "btn-default" class, 
Let's have a set of standard for naming.

File: docs/demo/general-form-elements.html (hunk 6 above)
Comment (erangatl) [inline docs/demo/general-form-elements.html]: -1 on having a 13 character prefix for ID field.
Let's have a set of standard for naming.

**Pull request merged**